plugins/controlboard/src/ControlBoardDriver.cpp

```diff
@@ -404,9 +433,15 @@ bool GazeboYarpControlBoardDriver::sendPositionsToGazebo(Vector &refs)
 bool GazeboYarpControlBoardDriver::sendPositionToGazebo(int j,double ref)
 {
     gazebo::msgs::JointCmd j_cmd;
-    prepareJointMsg(j_cmd,j,ref);
-    m_jointCommandPublisher->WaitForConnection();
-    m_jointCommandPublisher->Publish(j_cmd);
+
+    if(ref != m_oldReferencePositions[j])
+    {
+        // std::cout << "Sending new command: new ref is " << ref << "; old ref was " << m_oldReferencePositions[j] << std::endl;
```

Comment:
It is commented out so no noisy prints. I just kept in case the need of debugging should arise.